cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/DynamoDbTemplate.java

```diff
@@ -434,7 +434,7 @@ public void delete(final Item item, final PersistenceExceptionHandler<?>... pers
     private <T extends Item> Collection<T> executeQuery(final AttributeQuery query, final Class<T> itemClass) {
         final ItemConfiguration itemConfiguration = getItemConfiguration(itemClass);
 
-        if (!query.getCondition().containsNonNullOrEmptyValues()) {
+        if (!query.getCondition().containsRequiredComparisonValues()) {
```

Comment:
I was struggling for a name to be honest!  I think your suggestion is a good one.